tools/build_config/build_config.bzl

```diff
@@ -66,52 +65,37 @@ def build_config(
         ints: Build config field of type Int
         longs: Build config field of type longs
     """
+    build_config_file_path = "%s/src/main/java/BuildConfig.java" % (name)
 
-    build_config = "BuildConfig"
-    build_config_file_path = "src/main/%s/%s.kt" % (name, build_config)
+    build_config_generator = "@grab_bazel_common//tools/build_config:build_config_generator"
```

Comment:
Can we convert this to a dedicated action like we did for databinding stubs and avoid genrule invocation? 

Because otherwise I don't think we can use workers. 

https://github.com/grab/grab-bazel-common/blob/07ba02233cf9809c1601c2503d4a45c93a08ce88/tools/databinding/databinding_stubs.bzl#L158-L176

Reply:
genrule was used here to take advantage of being able to pass in `select()` within it's cmd to get it to expand at a later stage. 

With the dedicated action, we will not be able to have `select()` within the dict's value

Reply:
That's true, however when implementing an rule, we can merge `selects` as well, as described here https://bazel.build/docs/configurable-attributes#combining-selects.

So overall we can receive raw `dict` of values from `macro`, inside the macro implementation we can merge static and configurable params and pass it to a rule that consumes it similar to what's being done for [stubs](https://github.com/grab/grab-bazel-common/blob/07ba02233cf9809c1601c2503d4a45c93a08ce88/tools/databinding/databinding_stubs.bzl#L169).

Reply:
Updated to merge selects in new commit